web/src/components/tasks/TaxInput.tsx

```diff
@@ -106,6 +106,23 @@ export const TaxInput = (props: Props): ReactElement => {
     </div>
   );
 
+  const showNeedsAccountModalForGuest = (): (() => void) | undefined => {
```

Comment:
[dust/pebble] This made me do a double-take when I saw a function getting executing down on line 148, until I realized it's a function that returns a function.

That's fine, I see what we want it to be undefined or a function. But I think the naming should be revised to make it clear that this is NOT a function that "shows needs account modal for guest" - it returns a function that does that. Something like `getNeedsAccountFunction` or even `getHandleChangeOverride` would be more descriptive imo